lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,1890 @@
+        """
+        config = {}
+        remote_configuration_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_configuration_relation and remote_configuration_relation.app:
+            try:
+                raw_config = remote_configuration_relation.data[remote_configuration_relation.app][
+                    "alertmanager_config"
+                ]
+                if self._config_is_valid(json.loads(raw_config)):
+                    config = raw_config
+                else:
+                    logger.error("Config validation error.")
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no config has been provided."
+                )
+        return config
+
+    def templates(self) -> list:
+        """Returns Alertmanager templates sent inside the relation data bag.
+
+        If the `alertmanager-remote-configuration` relation exists and the relation data bag
+        contains Alertmanager templates, returns the templates in the form of a list.
+
+        Returns:
+            list: Alertmanager templates
+        """
+        templates = []
+        remote_configuration_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_configuration_relation and remote_configuration_relation.app:
+            try:
+                templates_raw = remote_configuration_relation.data[
+                    remote_configuration_relation.app
+                ]["alertmanager_templates"]
+                templates = json.loads(templates_raw)
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no templates have been provided."
+                )
+        return templates
+
+    @staticmethod
+    def _config_is_valid(config: dict) -> bool:
+        """Validates Alertmanager configuration.
+
+        Uses JSON Schema validator to check whether the Alertmanager configuration provided
+        by the `alertmanager-remote-configuration` consumer is valid.
+
+        Args:
+            config: Alertmanager configuration JSON
+
+        Returns:
+            bool: True/False depending on the configuration validity
+        """
+        try:
+            validate(instance=config, schema=ALERTMANAGER_CONFIG_JSON_SCHEMA)
+            return True
+        except exceptions.ValidationError:
+            return False
+
+    def _get_current_config(self) -> str:
+        """Gets current configuration of the Alertmanager.
+
+        Uses Alertmanager's API to get the current configuration.
+
+        Returns:
+            str: Alertmanager configuration
+        """
+        config_endpoint = "api/v2/status"
+        url = "{}/{}".format(self.api_address, config_endpoint)
+        response = requests.get(url)
+        return response.json()["config"]["original"] if response.status_code == 200 else ""
+
+
+class AlertmanagerRemoteConfigurationConsumer(Object):
```

Comment:
Could you please add a deployment scenario diagram to INTEGRATING.md?

Examples:
- https://github.com/canonical/grafana-agent-k8s-operator/blob/main/INTEGRATING.md#deployment-scenarios
- https://github.com/canonical/prometheus-scrape-config-k8s-operator/blob/main/INTEGRATING.md#deployment-scenarios

Reply:
I've updated `INTEGRATING.md`. It didn't have any diagrams, so I figured I'll just follow the style and added a written description.
Would that be ok?